lib/gtfs.ex

```diff
@@ -185,6 +190,31 @@ defmodule Gtfs do
     end
   end
 
+  @spec retrive_data(files(), boolean) :: t()
+  defp retrive_data(files, mocked_files?) do
+    if !mocked_files? && CacheFile.should_use_file?() do
+      Logger.info("Loading gfts data from cached file")
+
+      case CacheFile.load_gtfs() do
+        {:ok, state} ->
+          Logger.info("Loaded gtfs data from cached file")
+          state
+
+        _ ->
+          Logger.info("Loading gtfs data remote files")
+          state = parse_files(files)
+          Logger.info("Loaded gtfs data remote files")
+          CacheFile.save_gtfs(state)
+          state
+      end
+    else
+      Logger.info("Loading gtfs data remote files")
+      state = parse_files(files)
+      Logger.info("Loaded gtfs data remote files")
+      state
+    end
```

Comment:
- There's some duplicate code in these cases.
- What happens if the shape of the state has changed since the cache was saved?

Reply:
Re: changes to gtfs data, I was only thinking of this as a speed-up measure in development, so personally I'm happy just saying, "blow away your cache file anytime anything changes". I can write a custom mix task to actually do that, I had meant to do that and forgot.

Reply:
If there's been a change, can we reasonably detect it? Will loading the data into structs fail?

If so, perhaps we could handle a failed load by automatically removing the cache, and downloading and storing new data.

Reply:
I don't know, I know this area less well, but in my experience with Alerts I can say that small changes to this sort of data (e.g. a stop name changing) happen occasionally but would likely not be easily detected (would not fail to load into structs). I never experienced larger failures. Also, knowing that there was a cache that sped things up but that I should just delete if I even thought something might have changed did not feel particularly problematic. Worst case is the app takes longer to load once afterwards.

Reply:
Refatored out duplicate code.